mesonbuild/interpreter/type_checking.py

```diff
@@ -531,7 +531,10 @@ def _validate_win_subsystem(value: T.Optional[str]) -> T.Optional[str]:
 
 # Arguments exclusive to StaticLibrary. These are separated to make integrating
 # them into build_target easier
-_EXCLUSIVE_STATIC_LIB_KWS: T.List[KwargInfo] = []
+_EXCLUSIVE_STATIC_LIB_KWS: T.List[KwargInfo] = [
+    *[l.evolve(name=f'{l.name.split("_", 1)[0]}_static_args', since='1.3.0')
+      for l in _LANGUAGE_KWS],
```

Comment:
Why are we adding `<lang>_<static/shared>_args` to the static/shared library functions as well? Shouldn't this just be for the target types that are able to support both and have to differentiate?